cmd/influxd/backup/backup.go

```diff
@@ -121,8 +235,25 @@ func (cmd *Command) nextPath(path string) (string, error) {
 	}
 }
 
-// download downloads a snapshot from a host to a given path.
-func (cmd *Command) download(host string, m *snapshot.Manifest, path string) error {
+// downloadAndVerify will download either the metastore or shard to a temp file and then
+// rename it to a good backup file name after complete
+func (cmd *Command) downloadAndVerify(req *snapshotter.Request, path string) error {
+	tmppath := path + Suffix
+	if err := cmd.download(req, tmppath); err != nil {
+		return err
+	}
+
+	// Rename temporary file to final path.
+	if err := os.Rename(tmppath, path); err != nil {
+		return fmt.Errorf("rename: %s", err)
+	}
+
+	// TODO: Check file integrity.
```

Comment:
Does this `TODO` need to be done for this PR? Or should we track it as a separate issue?